src/requirements/data/exams/ExamCredit.ts

```diff
@@ -1,316 +1,443 @@
 /* eslint-disable max-len */
-import { ExamData } from '../../types';
+import { CourseTaken } from '../../types';
 
-const transferJSON: ExamData = {
+// const sampleExams: ExamsTaken = {
+//   AP: [
+//     {
+//       name: 'Computer Science A',
+//       score: 5,
+//     },
+//   ],
+//   IB: [
+//     {
+//       name: 'Chemical And Physical Systems',
+//       score: 5,
+//     },
+//   ],
+// };
+
+export type ExamRequirements = {
+  readonly name: string;
+  readonly fulfillment: {
+    readonly courseEquivalents: Record<string, number>;
+    readonly minimumScore: number;
+    readonly credits: number;
+    readonly majorsExcluded?: string[];
+  };
+};
+export type ExamData = Record<string, ExamRequirements[]>;
+
+export type ExamTaken = {
+  readonly name: string;
+  readonly score: number;
+};
+export type ExamsTaken = Record<string, ExamTaken[]>;
+
+function userDataToCourses(
+  college: string,
+  major: string,
+  userData: ExamsTaken,
+  examType: string
+): CourseTaken[] {
+  const userExams = userData[examType];
+  const exams = examData[examType];
+  const courses: CourseTaken[] = [];
+  userExams.forEach(userExam => {
+    // match exam to user-taken exam
+    let exam: ExamRequirements | undefined;
+    exams.forEach(e => {
+      // check if exam name matches and score is high enough
+      if (e.name.includes(userExam.name) && userExam.score >= e.fulfillment.minimumScore) {
+        // update exam variable if this exam has a higher minimum score
+        if (!exam || exam.fulfillment.minimumScore < e.fulfillment.minimumScore) {
+          exam = e;
+        }
+      }
+    });
```

Comment:
nit: you can do it functionally using `.find`

Reply:
How do I use `.find` with the second `if` guard? Like "find exam with max minimumScore compared to other exams"?
I had trouble with `.reduce` earlier but I got it to work now, would that resolve the nit? I totally agree with you btw, I was trying for a while to make it functional.

Reply:
oh didn't see the comparison stuff. then the code makes sense to me. not push blocking